src/Diagnostics.RuntimeHost/Controllers/DeploymentController.cs

```diff
@@ -0,0 +1,162 @@
+﻿using Diagnostics.Logger;
+using Diagnostics.ModelsAndUtils.Models;
+using Diagnostics.ModelsAndUtils.Models.Storage;
+using Diagnostics.RuntimeHost.Services;
+using Diagnostics.RuntimeHost.Services.CacheService;
+using Diagnostics.RuntimeHost.Services.StorageService;
+using Diagnostics.RuntimeHost.Services.DevOpsClient;
+using Diagnostics.RuntimeHost.Utilities;
+using Diagnostics.Scripts;
+using Diagnostics.Scripts.CompilationService.Utilities;
+using Diagnostics.Scripts.Models;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+
+namespace Diagnostics.RuntimeHost.Controllers
+{
+    [Authorize]
+    [Produces("application/json")]
+    [Route("api/deploy")]
+    public class DeploymentController : Controller
+    {
+
+        private IRepoClient devopsClient;
+        protected ICompilerHostClient _compilerHostClient;
+        protected IStorageService storageService;
+        private IInvokerCacheService detectorCache;
+
+        public DeploymentController(IServiceProvider services, IConfiguration configuration, IInvokerCacheService invokerCache)
+        {
+            this.storageService = (IStorageService)services.GetService(typeof(IStorageService));
+            this.devopsClient = (IRepoClient)services.GetService(typeof(IRepoClient));
+            this._compilerHostClient = (ICompilerHostClient)services.GetService(typeof(ICompilerHostClient));           
+            this.detectorCache = invokerCache;
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> Post([FromBody] DeploymentParameters deploymentParameters)
+        {
+            // If all required parameters are empty, reject the request.
+            if(string.IsNullOrWhiteSpace(deploymentParameters.CommitId) && string.IsNullOrWhiteSpace(deploymentParameters.FromCommitId)
```

Comment:
I am assuming there are other invalid combinations as well, like:
- only FromCommitId present
- only StartDate Present or only EndDate Present
- StartDate > EndDate etc
 
 I would suggest add a class Method in the DeploymentParameters Class only called Validate or something and have this logic there.

Reply:
For the first two in the list, I have handled them inside inner method that calls the devops API. I'll add validation for last one as well.